### PR TITLE
Docs: add a "Scaling out" hint to Quickstart.

### PR DESCRIPTION
**Pull request by concretevitamin**
Motivation: Current Quickstart does not give any pointers on how to go from the simple 1-task example to a more advanced scaling-out mode, which is where SkyPilot shines.

This PR adds a small section there as a hint rather than a complete guide.   I think it's important to give some impression right at Quickstart that we can do it.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below): rendered
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/getting-started/quickstart.rst [hunk 1]
```diff
@@ -178,5 +196,5 @@ Next steps:
 - Adapt :ref:`Tutorial: DNN Training <dnn-training>` to start running your own project on SkyPilot!
 - See the :ref:`Task YAML reference <yaml-spec>`, :ref:`CLI reference <cli>`, and `more examples <https://github.com/skypilot-org/skypilot/tree/master/examples>`_
 - To learn more, try out `SkyPilot Tutorials <https://github.com/skypilot-org/skypilot-tutorial>`_ in Jupyter notebooks
-- Try :ref:`Interactive Nodes <interactive-nodes>` -- launch VMs in one command without a YAML file
-- Explore SkyPilot's unique features in the rest of the documentation
+
+We invite you to explore SkyPilot's unique features in the rest of the documentation.
```
Comment (romilbhardwaj) [inline docs/source/getting-started/quickstart.rst:200]: Maybe we can leave this out, less fluff

File: docs/source/getting-started/quickstart.rst [hunk 2]
```diff
@@ -165,8 +165,26 @@ Stop/terminate a cluster
...
+
+Scaling out
+=========================
+
+So far, we have used the `serverful` CLIs of SkyPilot, where the notion of clusters is central.
+This is useful for development since it is convenient to log into a cluster and perform debugging.
+
+When you are ready to scale out (e.g., run 10s or 100s of jobs), SkyPilot supports two options:
+
+- Queue jobs on a cluster (see :ref:`Job Queue <job-queue>`); or
```
Comment (romilbhardwaj) [inline docs/source/getting-started/quickstart.rst:180]: The reader just used `sky exec` in the quickstart above. Perhaps we should reference it here, hinting that `sky exec` is how you submit jobs to the job queue
```suggestion
- Queue jobs on a cluster with :code:`sky exec` (see :ref:`Job Queue <job-queue>`); or
```

File: docs/source/getting-started/quickstart.rst [hunk 3]
```diff
@@ -165,8 +165,26 @@ Stop/terminate a cluster
...
+- Use :ref:`Managed Spot Jobs <spot-jobs>`, which is a `serverless` style API where clusters are not directly exposed.
+
+Managed spot jobs run on much cheaper spot instances, with automatic preemption recovery. Try it out with:
```
Comment (romilbhardwaj) [inline docs/source/getting-started/quickstart.rst:183]: preemption is a systems-y word, failure might be better understood
```suggestion
Managed spot jobs run on much cheaper spot instances, with automatic recovery from failures. Try it out with:
```

File: docs/source/getting-started/quickstart.rst [hunk 4]
```diff
@@ -165,8 +165,26 @@ Stop/terminate a cluster
...
-manage the lifecycle of clusters :ref:`here <interactive-nodes>`.
+terminate a cluster instead, run :code:`sky down mycluster`.
+
+Find more commands that manage the lifecycle of clusters in the :ref:`CLI reference <cli>`.
+
+Scaling out
+=========================
+
+So far, we have used the `serverful` CLIs of SkyPilot, where the notion of clusters is central.
+This is useful for development since it is convenient to log into a cluster and perform debugging.
```
Comment (romilbhardwaj) [inline docs/source/getting-started/quickstart.rst:176]: Isn't it also ~useful~ required for running jobs for on-demand clusters?

File: docs/source/getting-started/quickstart.rst [hunk 5]
```diff
@@ -165,8 +165,26 @@ Stop/terminate a cluster
...
+
+When you are ready to scale out (e.g., run 10s or 100s of jobs), SkyPilot supports two options:
+
+- Queue jobs on a cluster (see :ref:`Job Queue <job-queue>`); or
+- Use :ref:`Managed Spot Jobs <spot-jobs>`, which is a `serverless` style API where clusters are not directly exposed.
```
Comment (romilbhardwaj) [inline docs/source/getting-started/quickstart.rst:181]: Alternative to introducing `serverful` and `serverless` concepts
```suggestion
- Use :ref:`Managed Spot Jobs <spot-jobs>`, which runs jobs on spot instances and terminates instances on job completion
```

**Review by romilbhardwaj — COMMENTED** [review]
Thanks! Still debating whether we should introduce `serverful` and `serverless` concepts here, particularly since `serverless` makes me think about lambda functions.

File: docs/source/getting-started/quickstart.rst (hunk 3 above)
Comment (concretevitamin) [reply docs/source/getting-started/quickstart.rst:182]: I think it's ok to say preemptions; people who know "spot" should know about that word. "Failures" can mean user code failures which we cannot recover from.

File: docs/source/getting-started/quickstart.rst (hunk 5 above)
Comment (concretevitamin) [reply docs/source/getting-started/quickstart.rst]: Good call, reworded.

File: docs/source/getting-started/quickstart.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/getting-started/quickstart.rst]: Good call, reworded.

File: docs/source/getting-started/quickstart.rst (hunk 4 above)
Comment (concretevitamin) [reply docs/source/getting-started/quickstart.rst]: Reworded.

File: docs/source/getting-started/quickstart.rst (hunk 1 above)
Comment (concretevitamin) [reply docs/source/getting-started/quickstart.rst:199]: This seems like a reasonable tone to me. Points out "there are more unique features".

**Review by concretevitamin — COMMENTED** [review]
Agreed on dropping the "serverful/serverless" words. We don't need to risk introducing confusion. What this new section really meant to say is, "We support scale-out, here's two ways you can try."

**Review by romilbhardwaj — APPROVED** [review]
LGTM, thanks @concretevitamin!

**Pull request merged**